tests/Fixer/Import/NoUnusedImportsFixerTest.php

```diff
@@ -684,6 +684,136 @@ public function testWithSameNamespaceImportAndUnusedImportAfterNamespaceStatemen
         $this->doTest($expected, $input);
     }
 
+    public function testWhenImportedClassNameIsPrefixWithDashOfConstant()
+    {
+        $expected = <<<'EOF'
+<?php
+
+
+class Dummy
+{
+    const C = 'bar-bados';
+}
+EOF;
+
+        $input = <<<'EOF'
+<?php
+
+use Foo\Bar;
+
+class Dummy
+{
+    const C = 'bar-bados';
+}
+EOF;
+
+        $this->doTest($expected, $input);
+    }
+
+    public function testWhenImportedClassNameIsSuffixWithDashOfConstant()
+    {
+        $expected = <<<'EOF'
+<?php
+
+
+class Dummy
+{
+    const C = 'tool-bar';
+}
+EOF;
+
+        $input = <<<'EOF'
+<?php
+
+use Foo\Bar;
+
+class Dummy
+{
+    const C = 'tool-bar';
+}
+EOF;
+
+        $this->doTest($expected, $input);
+    }
+
+    public function testWhenImportedClassNameIsInsideWithDashOfConstant()
+    {
+        $expected = <<<'EOF'
+<?php
+
+
+class Dummy
+{
+    const C = 'tool-bar-bados';
+}
+EOF;
+
+        $input = <<<'EOF'
+<?php
+
+use Foo\Bar;
+
+class Dummy
+{
+    const C = 'tool-bar-bados';
+}
+EOF;
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @param string $expected
+     *
+     * @dataProvider provideWhenImportedClassIsUsedForConstantsCases
+     */
+    public function testWhenImportedClassIsUsedForConstants($expected)
+    {
+        $this->doTest($expected);
+    }
+
+    public function provideWhenImportedClassIsUsedForConstantsCases()
+    {
+        return array(
+            array(
+                '<?php
+use A\ABC;
+$a = 5-ABC::Test;
+$a = 5-ABC::Test-5;
+$a = ABC::Test-5;
+',
+            ),
+            array(
+                '<?php
+use A\ABC;
+$a = 5-ABC::Test;
+$a = 5-ABC::Test-5;
+',
+            ),
+            array(
+                '<?php
+use A\ABC;
+$a = 5-ABC::Test;
+',
+            ),
+            array('<?php
+use A\ABC;
+$a = ABC::Test-5;
+',
+            ),
+            array('<?php
+use A\ABC;
+$a = 5-ABC::Test-5;
+',
+            ),
+            array('<?php
+use A\ABC;
+$b = $a-->ABC::Test;
```

Comment:
I'm surprised this is valid PHP, what does this line do exactly?

Reply:
`$b` will be a boolean, the value is resolved by `$a--` than `>` than `ABC::Test`,
or throwing in some stuff to make it more easy;
`$b = $a-->ABC::Test;`
===
`$b = ($a--) > ABC::Test;`

Reply:
Thanks :)